charts/opentelemetry-collector/values.yaml

```diff
@@ -137,6 +133,17 @@ agentCollector:
 
   containerLogs:
     enabled: false
+    # Container runtime. One of `docker`, `cri-o`, or `containerd`
+    containerRunTime: ""
+    includeAgentLogs: false
+    enrichK8sMetadata: true
+    listOfAnnotations: []
+    listOfLabels: []
```

Comment:
K8s metadata applicable not only to logs, but widely used for traces and metrics as well.

All the default configuration can be easily overridden (by merging with defaults) in user's values.yaml. So I don't see a lot of benefit in providing extra configuration ways. Especially `listOfAnnotations` and `listOfLabels` can be easily replaced by the existing configuration method.
```
config:
  processors:
    k8s_tagger:
      annotations: ...
      labels: ...
```
it's just one line more to print than
```
agentCollector:
  containerLogs:
    listOfAnnotations:
    listOfLabels:
```
but less configuration options -> less confusion.

Speaking of `enrichK8sMetadata`, I would recommend to move the option one level up to be able to apply it to any of the pipelines. Like:
```
agentCollector:
  pipelinesWithK8sMetadata: [metrics, traces, logs]
```

Let's keep `agentCollector.k8sMetadataPipelines: []` by default.

@tigrannajaryan @naseemkullah thoughts?

Reply:
> K8s metadata applicable not only to logs, but widely used for traces and metrics as well.
> 
> All the default configuration can be easily overridden (by merging with defaults) in user's values.yaml. So I don't see a lot of benefit in providing extra configuration ways. Especially `listOfAnnotations` and `listOfLabels` can be easily replaced by the existing configuration method.
> 
> ```
> config:
>   processors:
>     k8s_tagger:
>       annotations: ...
>       labels: ...
> ```
> 
> it's just one line more to print than
> 
> ```
> agentCollector:
>   containerLogs:
>     listOfAnnotations:
>     listOfLabels:
> ```
> 
> but less configuration options -> less confusion.

Good catch 💯 

> Speaking of `enrichK8sMetadata`, I would recommend to move the option one level up to be able to apply it to any of the pipelines. Like:
> 
> ```
> agentCollector:
>   pipelinesWithK8sMetadata: [metrics, traces, logs]
> ```
> 
> Let's keep `agentCollector.k8sMetadataPipelines: []` by default.
> 

I just learned about this. so we would have both `agentCollector.pipelines` and `agentCollector.pipelinesWithK8sMetadata`?

Seeing that this is a helm chart maybe we just add the `k8s_tagger` and so on directly in the pipelines out of the box if that makes sense?

Reply:
> I just learned about this. so we would have both agentCollector.pipelines and agentCollector.pipelinesWithK8sMetadata?

right

> Seeing that this is a helm chart maybe we just add the k8s_tagger and so on directly in the pipelines out of the box if that makes sense?

Sounds good, but having `k8s_tagger` enabled by default requires installing the cluster role. At least we need a way to opt out from that.

Reply:
> but less configuration options -> less confusion.

I agree.

Reply:
I think by having less config, you are making the users to study more on documentation to properly set up configOverride. 
For example, 
```
      k8s_tagger:
        passthrough: false
        auth_type: "kubeConfig"
        pod_association:
          - from: resource_attribute
            name: k8s.pod.uid
        extract:
          metadata:
            - deployment
            - cluster
            - namespace
            - node
            - startTime
          annotations: []
          labels: []
        filter:
          node_from_env_var: KUBE_NODE_NAME
```
this is required config for enriching it with k8s metadata. 
most of it is static. Users doesn't need to know what `passthrough` value they need to pick, nor search though this plugin documentation to properly choose what `pod_association` I need to use. 
So I think in this case, "less configuration options -> less confusion" doesn't quite fit here. it is more like "more guided helm chart -> less confusion". 
We are helping users by presenting the configuration they need to look and hiding what they don't need to waste their minds on.